server/nethttp/handler.go

```diff
@@ -1,25 +1,170 @@
+			return
+		}
+		if r.URL.Path == "/login" { // TODO: do not check auth here, but do it in all other routes
+			h.Login(w, r)
+			return
+		}
+		if r.URL.Path == "/sendMessage" {
+			h.SendMessage(w, r)
+			return
+		}
+		err = object.ErrNotFound
+	} else {
+		err = object.ErrMethodNotAllowed
+	}
+
+	if err != nil {
+		h.Error(w, r, err)
+	}
+}
+
+// Error is a handler for errors, returns error in JSON
+func (h Handler) Error(w http.ResponseWriter, r *http.Request, err error) {
+	log.Println("usual error:", err, r.Method, r.URL.Path)
+
+	statusCode := 500
+
+	// error handling. TODO: think about how to implement this (LocalError{Code, GlobalError}????)
+	switch err {
+	case object.ErrNotFound:
+		statusCode = 404
+	case object.ErrTakenUsername:
+		statusCode = 409
+	case object.ErrUsernameNotSpecified, object.ErrWrongID, object.ErrPassNotSpecified, object.ErrDataNotSpecified:
+		statusCode = 400
+	}
+
+	var resp struct {
+		Error string `json:"error"`
+	}
+
+	// write error message to response
+	resp.Error = err.Error()
+	w.WriteHeader(statusCode)
+
+	err = json.NewEncoder(w).Encode(resp)
+	if err != nil {
+		log.Println("CRITICAL ERROR:", err, r.Method, r.URL.Path)
+	}
+}
+
+// GetUser is a handler for GET /getUser, requests user id, returns user's info in JSON
+func (h Handler) GetUser(w http.ResponseWriter, r *http.Request) {
+	h.Error(w, r, object.ErrNotImplemented)
+}
+
+// GetMessage is a handler for GET /getMessage, requests message id, returns message's info in JSON
+func (h Handler) GetMessage(w http.ResponseWriter, r *http.Request) {
+	h.Error(w, r, object.ErrNotImplemented)
+}
+
+// Register is a handler for POST /register, requests user's username and password, returns user's info in JSON
+func (h Handler) Register(w http.ResponseWriter, r *http.Request) {
+	// get User from request
+	var user object.User
+	err := json.NewDecoder(r.Body).Decode(&user)
+	if err != nil {
+		h.Error(w, r, err)
+		return
+	}
+
+	if user.Username == "" && user.Password == "" {
+		h.Error(w, r, object.ErrDataNotSpecified)
+		return
+	}
+	if user.Username == "" {
+		h.Error(w, r, object.ErrUsernameNotSpecified)
+		return
+	}
+	if user.Password == "" {
+		h.Error(w, r, object.ErrPassNotSpecified)
+		return
+	}
+
+	id, err := h.Service.SaveUser(user)
+	if err != nil {
+		h.Error(w, r, err)
+		return
+	}
+
+	user.ID = id
+	// write response
+	w.WriteHeader(200)
+	err = json.NewEncoder(w).Encode(user)
+	if err != nil {
+		h.Error(w, r, err)
+		return
 	}
 }
 
-// ServeHTTP is a root method for http handlers
-func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {}
+// Login is a handler for POST /login, requests user's username and password, returns user's info in JSON
+func (h Handler) Login(w http.ResponseWriter, r *http.Request) {
+	h.Error(w, r, object.ErrNotImplemented)
+}
+
+// SendMessage is a handler for POST /sendMessage, requests a token, returns a message in JSON
+func (h Handler) SendMessage(w http.ResponseWriter, r *http.Request) {
```

Comment:
```suggestion
func (h Handler) PostMessage(w http.ResponseWriter, r *http.Request) {
```
You shall probably keep the convention you've started with your GET methods:
Get /something -> GetSomething()
Post /something -> PostSomething()

Reply:
done